firefox-ios/Client/Frontend/Library/Bookmarks/BookmarksViewController.swift

```diff
@@ -753,21 +775,10 @@ extension BookmarksViewController: LibraryPanelContextMenu {
         }).items
         var actions: [PhotonRowActions] = [editBookmark] + defaultActions
 
-        let pinTopSite = SingleActionViewModel(title: .AddToShortcutsActionTitle,
-                                               iconString: StandardImageIdentifiers.Large.pin,
-                                               tapHandler: { _ in
-            self.profile.pinnedSites.addPinnedTopSite(site).uponQueue(.main) { result in
-                if result.isSuccess {
-                    SimpleToast().showAlertWithText(.LegacyAppMenu.AddPinToShortcutsConfirmMessage,
-                                                    bottomContainer: self.view,
-                                                    theme: self.currentTheme())
-                } else {
-                    self.logger.log("Could not add pinned top site",
-                                    level: .debug,
-                                    category: .library)
-                }
-            }
-        }).items
+        let pinTopSite = site.isPinnedSite
```

Comment:
Couldn't we do directly the following ?
```suggestion
        let pinTopSiteAction = createPinUnpinAction(for: site, isPinned: site.isPinnedSite)
```

Reply:
Wow! that I must have lost it in chain of thoughts haha!

Reply:
🤣 no problem can happen